backend/api/event/views.py

```
@@ -427,3 +491,107 @@ def get_event_details(request):
+    # Get current user ID from cookies
+    aget_session = sync_to_async(get_session)
+    session = None
+    user_id = -1
+    if ACCOUNT_COOKIE_NAME in request.COOKIES:
+        session = await aget_session(request.COOKIES[ACCOUNT_COOKIE_NAME])
+    if session is None and GUEST_COOKIE_NAME in request.COOKIES:
+        session = await aget_session(request.COOKIES[GUEST_COOKIE_NAME])
+    if session:
+        user_id = session.user_account_id
+
+    async def event_generator():
```

Ruff (0.15.18)

[warning] Missing return type annotation for private function `event_generator`

(ANN202)